src/nfa/operations.cc

```diff
@@ -56,6 +56,126 @@ namespace {
         return lts_for_simulation.compute_simulation();
     }
 
+    bool vec_contains(std::vector<std::pair<State, State>> vec, std::pair<State, State> par){
+        for (size_t i = 0; i < vec.size(); i++){
+            if (vec[i] == par){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    Nfa reduce_testing(const Nfa& aut){
+        // ! Preprocessing
+        Nfa result_nfa;
+        Nfa reverted_nfa;
+
+        std::vector<std::pair<State, State>> result_sim {}; // R
```

Comment:
instead of `std::vector`, some other data structure with faster lookup would be better (sparse bitmap/`std::set`/`std::unordered_set`, ...)